hazelcast/src/main/java/com/hazelcast/instance/ServerSocketHelper.java

```diff
@@ -122,6 +125,10 @@ static ServerSocketChannel openServerSocketChannel(EndpointQualifier qualifier,
             serverSocket.setReuseAddress(reuseAddress);
             serverSocket.setSoTimeout(SOCKET_TIMEOUT_MILLIS);
```

Comment:
It would be nice to have this setting based on the endpoint config, but its out of the scope and risky to introduce changes here at this stage.